### PR TITLE
snapcraft: Add support for sideloaded lxd-agent debug executable

### PR DESCRIPTION
**Pull request by tomponline**
Signed-off-by: Thomas Parrott <tomp@tomp.uk>

### CODE REVIEW

File: snapcraft/commands/daemon.start [hunk 1]
```diff
@@ -189,6 +189,12 @@ if [ "${lvm_external:-"false"}" = "true" ]; then
     ln -s "${SNAP}/wrappers/run-host" "/run/bin/lvresize"
 fi
 
+# Detect presence of sideloaded lxd-agent executable.
+if [ -x "/var/snap/lxd/common/lxd-agent.debug" ]; then
```
Comment (stgraber) [inline snapcraft/commands/daemon.start]: This should be `"${SNAP_COMMON}/lxd-agent.debug"`

File: snapcraft/commands/daemon.start [hunk 2]
```diff
@@ -189,6 +189,12 @@ if [ "${lvm_external:-"false"}" = "true" ]; then
...
+    echo "==> WARNING: Using a custom debug lxd-agent binary!"
+    ln -s "/var/snap/lxd/common/lxd-agent.debug" "/run/bin/lxd-agent"
```
Comment (stgraber) [inline snapcraft/commands/daemon.start]: same here

File: snapcraft/commands/daemon.start (hunk 1 above)
Comment (tomponline) [reply snapcraft/commands/daemon.start]: OK cool. I checked initially, and saw quite a few places that reference `/var/snap/lxd/common/` in that file so thought that was the way to do it.

File: snapcraft/commands/daemon.start (hunk 2 above)
Comment (tomponline) [reply snapcraft/commands/daemon.start]: Fixed

File: snapcraft/commands/daemon.start (hunk 1 above)
Comment (tomponline) [reply snapcraft/commands/daemon.start]: Fixed

**Pull request merged**